File: prototype/rails/src/test/delivery/mail_magic_link_delivery_test.rb
require "test_helper"

class MailMagicLinkDeliveryTest < ActiveSupport::TestCase
  test "it refuses to send until email is wired up" do
    error = assert_raises(NotImplementedError) do
      MailMagicLinkDelivery.new.deliver(email: "artist@example.com", url: "http://localhost:3000/auth/magic/abc")
    end

    assert_equal "Email delivery is not implemented yet", error.message
  end
end

File: prototype/node/src/app/delivery/mail-magic-link-delivery.test.ts
import { test } from 'node:test'
import assert from 'node:assert/strict'
import { NotImplementedError } from '../not-implemented-error.ts'
import { mailMagicLinkDelivery } from './mail-magic-link-delivery.ts'

test('it refuses to send until email is wired up', () => {
  assert.throws(
    () =>
      mailMagicLinkDelivery.deliver({
        email: 'artist@example.com',
        url: 'http://localhost:4000/auth/magic/abc',
        actorType: 'seller',
      }),
    new NotImplementedError('Email delivery is not implemented yet'),
  )
})
